app/controllers/checkout_controller.rb

```diff
@@ -261,6 +242,13 @@ def update_failed_response
     end
   end
 
+  def rescue_from_spree_gateway_error(error)
+    flash[:error] = t(:spree_gateway_error_flash_for_checkout, error: error.message)
+    # This can also happen during the edit action
```

Comment:
IMO we should stick to the `rescue_from` defined at line 35 for both `#edit` and `#update` or the rescue block but having separate approaches makes things harder. This PR is a good chance.

Reply:
I removed this comment and renamed the update_failed method to action_failed. I think that makes it more clear. It's just a generic method, not specific to edit or update.

But this takes me to another detail of this method.
The specific rescue in the update action that calls this method was added by Maikel here:
https://github.com/openfoodfoundation/openfoodnetwork/commit/bd67d63b63cca3c889755d95fcf1dd94a16ae9e0
Maybe it's a mistake because this checkout controller already had and still has the generic:
 `rescue_from Spree::Core::GatewayError, with: :rescue_from_spree_gateway_error`
Maybe we can just remove this rescue from the update action? I think we can. I am just asking because this is such a critical part of the checkout process.

Reply:
> Maybe we can just remove this rescue from the update action?

I believe we can, otherwise, Rails would be broken

Reply:
ah, I think I see why we need this now! If I remove the first rescue the second more generic rescue will be triggered and the rescue_from rule will not be used as we want. This is why Maikel added it I am pretty sure, I think I'll add a comment to explain this :+1:

Reply:
I ended up removing the rescue_from line and adding the rescue to the edit action. I think this makes it very simple:
- edit rescues GatewayError
- update rescues GatewayError and StandardError